src/DIRAC/MonitoringSystem/Client/DataOperationSender.py

```diff
@@ -41,16 +41,27 @@ def __init__(self):
             self._sendDataMethods.append(getattr(self, f"_sendData{backend}"))
             self._commitMethods.append(getattr(self, f"_commit{backend}"))
 
-    def _sendDataMonitoring(self, baseDict, commitFlag=False, delayedCommit=False, startTime=False, endTime=False):
+    def _sendDataMonitoring(
+        self, baseDict, commitFlag=False, delayedCommit=False, startTime=False, endTime=False, failedRecords=None
+    ):
         """Send the data to the monitoring system"""
-        # Some fields added here are not known to the Accounting, so we have to make a copy
-        monitoringDict = copy.deepcopy(baseDict)
```

Comment:
We do not have compound object, so `copy` is much faster

Reply:
So Monitoring was already not updating the dictionary, right? What was leading to the crash then?

Reply:
Ah I guess you are confused by the commit message (rightly so). 
It's because I originally did something a bit different in the first commit so I will reword it

Reply:
Thanks!